pbcore/io/FastqIO.py

```diff
@@ -72,7 +72,7 @@ def __init__(self, header, sequence, quality=None, qualityString=None):
             # Only one of quality, qualityString should be provided
             assert (quality is None) != (qualityString is None)
             if quality is not None:
-                self._quality = quality
+                self._quality = np.array(quality).astype(np.uint8)
```

Comment:
I'd prefer if you would just check the argument type and raise an exception if it's not a numpy array of the desired type. Existing calling code within pbcore already calls this function with the correct type---the code as written here will perform 2 additional copies of a potentially large array.